traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -844,22 +849,23 @@ func updateV31(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, dsV31 *
 		return nil, http.StatusInternalServerError, nil, errors.New("getting delivery service type during update: " + err.Error())
 	}
 
-	oldCDNName := ""
-	oldRoutingName := ""
 	errCode := http.StatusOK
 	var userErr error
 	var sysErr error
+	var oldDetails TODeliveryServiceOldDetails
 	if dsType.HasSSLKeys() {
-		oldCDNName, oldRoutingName, userErr, sysErr, errCode = getOldDetails(*ds.ID, tx)
+		oldDetails, userErr, sysErr, errCode = getOldDetails(*ds.ID, tx)
 		if userErr != nil || sysErr != nil {
 			return nil, errCode, userErr, sysErr
 		}
 		exists, err := getSSLVersion(*ds.XMLID, tx)
 		if err != nil {
 			return nil, http.StatusInternalServerError, nil, fmt.Errorf("querying delivery service with sslKeyVersion failed: %s", err)
 		}
-		if exists && (oldCDNName != *ds.CDNName || oldRoutingName != *ds.RoutingName) {
-			return nil, http.StatusBadRequest, errors.New("delivery service has ssl keys that cannot be automatically changed, therefore CDN and routing name are immutable"), nil
+		if ds.CDNName != nil && ds.RoutingName != nil {
+			if exists && (oldDetails.OldCdnName != *ds.CDNName || oldDetails.OldRoutingName != *ds.RoutingName) {
```

Comment:
i know u didn't create this variable but think you can change `exists` to `sslKeysExist` to make it semantic?